WalletWasabi.Gui/Controls/NoparaPasswordBox.cs

```diff
@@ -315,9 +321,10 @@ private bool OnTextInput(string text, bool isPaste)
 				SelectionStart = SelectionEnd = CaretIndex = 0;
 				_supressChanges = false;
 			}
-			if (isPaste && Sb.Length + text.Length > Constants.MaxPasswordLength) // Do not allow pastes that would be too long
+			if (Sb.Length + text.Length > Constants.MaxPasswordLength) // Do not allow insert that would be too long.
 			{
 				handledCorrectly = false;
+				_ = DisplayWarningAsync(PasswordTooLongMessage);
```

Comment:
Any way to avoid this being called asynchronously?

Reply:
We are doing the same elsewhere so let it be, this will be replaced by notifications.